src/openqaoa-core/backends/wrapper.py

```diff
@@ -0,0 +1,200 @@
+    round_value,
+)
+
+
+class BaseWrapper(VQABaseBackend):
+    def __init__(self, backend):
+        self.backend = backend
+
+    def __getattr__(self, name):
+        return getattr(self.backend, name)
+
+    def expectation(self, *args, **kwargs):
+        return self.backend.expectation(*args, **kwargs)
+
+    def expectation_w_uncertainty(self, *args, **kwargs):
+        return self.backend.expectation_w_uncertainty(*args, **kwargs)
+
+    def exact_solution(self, *args, **kwargs):
+        return self.backend.exact_solution(*args, **kwargs)
+
+
+class SPAMTwirlingWrapper(BaseWrapper):
+    def __init__(self, backend, n_batches, calibration_data_location):
+        super().__init__(backend)
+        self.n_batches = n_batches
+        self.calibration_data_location = calibration_data_location
+
+        with open(self.calibration_data_location, "r") as f:
+            calibration_data = json.load(f)
+
+        if (
+            self.calibration_data_location
+            == "calibration_data/8bc00556-2824-4d38-b563-706a803c6401--calibration.json"
+            or self.calibration_data_location
+            == "calibration_data/ac469846-25e7-40be-b0d0-2d60ae29d973--calibration.json"
+        ):
+            # Rigetti's data:
+            out = calibration_data["results"]["measurement_outcomes"]
+
+            # convert list to strings
+            calibration_measurements = {}
+            for outcomes in out:
+                for state, counts in outcomes.items():
+                    state = "".join([str(e) for e in eval(state)])
+                    if calibration_measurements.get(state) is None:
+                        calibration_measurements.update({state: counts})
+                    else:
+                        calibration_measurements[state] += counts
+        else:
+            calibration_measurements = calibration_data["results"][
+                "measurement_outcomes"
+            ]
+
+        calibration_registers = calibration_data["register"]
+
+        qubit_mapping = self.backend.initial_qubit_mapping
+
+        self.calibration_factors = calculate_calibration_factors(
+            self.backend.qaoa_descriptor.cost_hamiltonian,
+            calibration_measurements,
+            calibration_registers,
+            qubit_mapping,
+        )
+
+        """
+        This doesn't work because keys are tuples  =(
+        # save calibration factors just in case
+        with open('test.json', "w") as fp:
+            json.dump(self.calibration_factors, fp)
+        """
+
+    def get_counts(self, params, n_shots=None, seed=42):
+        """
+        Overrides the get_counts function of the backend object.
+        Modified function to...
+            divide into batches
+            change the self.append_state according to the schedule, s.
+            get the counts and classically negate them
+            combine all batches into a count dict under BFA
+        """
+        random.seed(seed)
```

Comment:
This needs to be a bit documented. Also, I am not sure whether fixing the seed all of a sudden will have unforeseen side effects across the code.

I am not saying that this is wrong, but it may be dangerous!

Reply:
See https://stackoverflow.com/questions/52544935/best-way-to-revert-to-a-random-seed-after-temporarily-fixing-it

Reply:
For now, it's been changed to. 

if seed is not None:
            random.seed(seed)

Have to decide if we want to create a new instance, and if this should be done not only here but everywhere we set a seed.